apps/src/javalab/JavalabEditor.jsx

```diff
@@ -93,28 +112,34 @@ class JavalabEditor extends React.Component {
     });
   }
 
-  makeListeners(key) {
-    return {
-      onContextMenu: e => {
-        this.openTabContextMenu(key, e);
-      }
-    };
+  onChangeTabs(key) {
+    if (key !== this.state.activeTabKey) {
+      this.setState({
+        showMenu: false,
+        contextTarget: null,
+        activeTabKey: key
+      });
+    }
   }
 
-  openTabContextMenu(key, e) {
-    e.preventDefault();
-    const boundingRect = e.target.getBoundingClientRect();
-    this.setState({
-      showMenu: true,
-      contextTarget: key,
-      menuPosition: {
-        top: `${boundingRect.bottom}px`,
-        left: `${boundingRect.left}px`
-      }
-    });
+  toggleTabMenu(key, e) {
+    if (key === this.state.contextTarget) {
```

Comment:
NIT: I'd recommend adding a couple comments here and the other places. i.e. I was pretty confused if the tabs were "tab navigation" or "editor tabs" and whether the keys were "ids" or "keyboard hotkeys." :)